### PR TITLE
feat(CX-2942): create median sale price screen wrapper

### PR DESCRIPTION
**Pull request by MounirDhahri**
The type of this PR is: **feature**

This PR solves [CX-2942]

### Description

- This PR adds the median sale price at auction screen wrapper.

**Note:**
- It's still not yet clear if we will hide or show the navigation and footer.
- The medium type will be sent as a query param, this is in order to support deep linking to the same page if needed later in App. When none is specified or a wrong one is specified, the fallback should be ideally `All`, as we do in App.

https://user-images.githubusercontent.com/11945712/196888467-a6e0c66e-9ffe-4bed-88f9-d958a12fab72.mov

[CX-2942]: https://artsyproduct.atlassian.net/browse/CX-2942?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Apps/MyCollectionInsights/myCollectionInsightsRoutes.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import loadable from "@loadable/component"
+import { AppRouteConfig } from "System/Router/Route"
+
+const MyCollectionInsightsMedianSalePriceAtAuction = loadable(
+  () =>
+    import(
+      /* webpackChunkName: "myCollectionInsightsBundle" */ "./Routes/MyCollectionInsightsMedianSalePriceAtAuction/MyCollectionInsightsMedianSalePriceAtAuction"
```
Comment (MounirDhahri) [inline src/Apps/MyCollectionInsights/myCollectionInsightsRoutes.tsx:7]: Does anyone know how I can generate this? using the `build` command maybe 👀?

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Routes added to `routes.tsx` should have a corresponding cypress.js smoke test. See the `cypress/e2e` folder for examples.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against e1367b09c4b52959378508d6fdc63ff789b03a52
</p>

File: src/Apps/MyCollectionInsights/Routes/MyCollectionInsightsMedianSalePriceAtAuction/MyCollectionInsightsMedianSalePriceAtAuction.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { Box, Text } from "@artsy/palette"
+
+export const MyCollectionInsightsMedianSalePriceAtAuction = () => {
+  return (
+    <Box>
+      <Text mt={2} variant={"lg-display"}>
```
Comment (MrSltun) [inline src/Apps/MyCollectionInsights/Routes/MyCollectionInsightsMedianSalePriceAtAuction/MyCollectionInsightsMedianSalePriceAtAuction.tsx:6]: **Issue:** According to the designs, the header should be `lg` in Desktop view and `lg-display` in mWeb
```suggestion
      <Text mt={2} variant={["lg-display", "lg"]}>
```

**Review by MrSltun — COMMENTED** [review]
Great work Mounir! 🌟 I just left a few comments
And could you please add an e2e test for the new route?

**Comment by MounirDhahri** [comment]
@MrSltun Yes, I created a ticket to add some smoke tests for the routes since it's still empty now 🙏

**Comment by MrSltun** [comment]
@MounirDhahri In that case I think we can merge it now

**Review by MrSltun — APPROVED** [review]

File: src/Apps/Settings/Routes/Insights/Components/__tests__/InsightsMedianSalePrice.jest.tsx [hunk 3]
```diff
@@ -20,23 +32,55 @@ describe("InsightsMedianSalePrice", () => {
...
-      expect(screen.getByText("American, 1928–1987")).toBeInTheDocument()
-      expect(screen.getByText("Print")).toBeInTheDocument()
-      expect(screen.getByText("US$39,930")).toBeInTheDocument()
+      const artistRow = screen.getByText("Takashi Murakami")
+
+      fireEvent.click(artistRow)
+
+      expect(mockPush).toHaveBeenCalledWith(
+        "/my-collection/median-sale-price-at-auction/takashi-murakami-id?medium=Print"
+      )
```
[the remark this answers is not in the document]
Comment (lordkiz) [reply src/Apps/Settings/Routes/Insights/Components/__tests__/InsightsMedianSalePrice.jest.tsx:83]: 👏

**Review by lordkiz — APPROVED** [review]
LGTM
